azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessagingFactory.java

```diff
@@ -300,32 +294,40 @@ private void onReactorError(Exception cause) {
         if (!this.open.isDone()) {
             this.onOpenComplete(cause);
         } else {
-            final Connection currentConnection = this.connection;
+            if (this.getIsClosingOrClosed()) {
+                return;
+            }
+
+            TRACE_LOGGER.warn(String.format(Locale.US, "onReactorError messagingFactory[%s], hostName[%s], error[%s]",
+                    this.getClientId(), this.getHostName(),
+                    cause.getMessage()));
+
+            final Connection oldConnection = this.connection;
+            final List<Link> oldRegisteredLinksCopy = new LinkedList<>(this.registeredLinks);
 
             try {
-                if (this.getIsClosingOrClosed()) {
-                    return;
-                } else {
-                    this.startReactor(new ReactorHandler());
-                }
+                TRACE_LOGGER.info(String.format(Locale.US, "onReactorError messagingFactory[%s], hostName[%s], message[%s]",
+                        this.getClientId(), this.getHostName(),
+                        "starting new reactor"));
+
+                this.startReactor(new ReactorHandlerWithConnection());
             } catch (IOException e) {
                 TRACE_LOGGER.error(String.format(Locale.US, "messagingFactory[%s], hostName[%s], error[%s]",
                         this.getClientId(), this.getHostName(),
                         ExceptionUtil.toStackTraceString(e, "Re-starting reactor failed with error")));
 
-                this.onReactorError(cause);
+                throw new RuntimeException(e);
```

Comment:
>throw new RuntimeException(e); [](start = 16, length = 30)

Why are we throwing here?
To understand if this is intended - the sideeffect of this is - this wil not bubble up the error to the User. An internal Thread in ExecutorService will abort with this error.